modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java

```diff
@@ -0,0 +1,506 @@
+                        cameraProfile.setFeatureCode(PluginConstants.OperationCodes.CAMERA);
+                        cameraProfile.setData(item.getData());
+                        if (item.getData().equals(PluginConstants.SyncML.SYNCML_DATA_ONE)) {
+                            cameraProfile.setEnable(true);
+                        } else {
+                            cameraProfile.setEnable(false);
+                        }
+                        profiles.add(cameraProfile);
+                    }
+                    if (item.getSource().getLocURI().equals(info.getCode()) && info.name().equals(
+                            PluginConstants.OperationCodes.ENCRYPT_STORAGE_STATUS)) {
+                        Profile encryptStorage = new Profile();
+                        encryptStorage.setFeatureCode(PluginConstants.OperationCodes.ENCRYPT_STORAGE);
+                        encryptStorage.setData(item.getData());
+                        if (item.getData().equals(PluginConstants.SyncML.SYNCML_DATA_ONE)) {
+                            encryptStorage.setEnable(true);
+                        } else {
+                            encryptStorage.setEnable(false);
+                        }
+                        profiles.add(encryptStorage);
+                    }
+                    if (item.getSource().getLocURI().equals(info.getCode()) && info.name().equals(
+                            PluginConstants.OperationCodes.DEVICE_PASSWORD_STATUS)) {
+                        Profile encryptStorage = new Profile();
+                        encryptStorage.setFeatureCode(PluginConstants.OperationCodes.PASSCODE_POLICY);
+                        encryptStorage.setData(item.getData());
+                        if (item.getData().equals(PluginConstants.SyncML.SYNCML_DATA_ZERO)) {
+                            encryptStorage.setEnable(true);
+                        } else {
+                            encryptStorage.setEnable(false);
+                        }
+                        profiles.add(encryptStorage);
+                    }
+                    if (!item.getData().isEmpty() && item.getSource().getLocURI().equals(lockUri)) {
+                        String pinValue = item.getData();
+                        NotificationManagementService nmService = WindowsAPIUtils.getNotificationManagementService();
+                        Notification notification = new Notification();
+                        notification.setDescription(pinValue);
+                        notification.setOperationId(result.getCommandReference());
+                        notification.setDeviceIdentifier(deviceIdentifier);
+                        try {
+                            nmService.addNotification(notification);
+                        } catch (NotificationManagementException e) {
+                            String msg = "Failure Occurred in getting notification service.";
+                            log.error(msg);
```

Comment:
Avoid using "log and throw" as much as possible, as it can clutter the logs thereby making things difficult to read them.